File: CPP/small_codes/file_handling.cpp
#include <fstream>
#include <iostream>
using namespace std;

int main(){
    ofstream out("data.txt"); out<<"Hello, World! I am Souvik and I am learning C++.\n"; out.close();
    ifstream in("data.txt"); string line; getline(in,line);
    cout<<line<<endl; in.close(); return 0;
}

File: Java/small_codes/FileHandling.java
import java.io.*;

public class FileHandling {
    public static void main(String[] args) {
        String filename = "example.txt";
        String content = "Hello, this is Souvik Pramanik.!\n";

        // Write to file
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(filename))) {
            writer.write(content);
            writer.write("I am taking another line to say I am starting to learn Java.");
            System.out.println("File written successfully.");
        } catch (IOException e) {
            System.out.println("Error writing file: " + e.getMessage());
        }

        // Read from file
        try (BufferedReader reader = new BufferedReader(new FileReader(filename))) {
            String line;
            System.out.println("\nReading file content:");
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }
    }
}
